src/components/widgets/TableWidget/components/Header.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useCallback, useEffect } from 'react'
+import * as styles from '../TableWidget.less'
+import Select from '../../../ui/Select/Select'
+import ActionLink from '../../../ui/ActionLink/ActionLink'
+import { useTranslation } from 'react-i18next'
+import { useDispatch, useSelector } from 'react-redux'
+import { Store } from '../../../../interfaces/store'
+import { parseFilters } from '../../../../utils/filters'
+import { $do } from '../../../../actions/actions'
+
+export interface HeaderProps {
+    bcName: string
+    widgetName: string
+}
+
+function Header({ bcName, widgetName }: HeaderProps) {
+    const { t } = useTranslation()
+    const [filterGroupName, setFilterGroupName] = React.useState(null)
+
+    const { filtersExist, limitBySelf, filterGroups, cursor } = useSelector((store: Store) => {
+        const bc = store.screen.bo.bc[bcName]
+        const bcCursor = bc?.cursor
+        const filters = store.screen.filters[bcName]
+
+        return {
+            filtersExist: filters?.length,
```

Comment:
You might want to convert into boolean here
1) Your effect only cares for false value, but will fire on each change in length
2) In render you checking it like ` {filtersExist && ... }`, if filters array is empty the number "0" will be shown in UI

Reply:
Thank you. I fixed it.